src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String COMMA = ",";
+    private static final String SEPARATOR = System.lineSeparator();
+    private static final String BUYING_OPERATION = "buy";
+    private static final String SUPPLY_OPERATION = "supply";
+    private static final String RESULT = "result";
+    private static final int NUMBER_OF_PARTS = 2;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String fileContent = readFromFile(fromFileName);
+        String report = makeReport(fileContent);
+        writeToFile(toFileName,report);
+    }
+
+    private String readFromFile(String fileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            String value;
+            while ((value = bufferedReader.readLine()) != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from file" + fileName, e);
+        }
+        return stringBuilder.toString();
+    }
+
+    private String makeReport(String data) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+        String[] lines = data.split(System.lineSeparator());
+        for (String line : lines) {
+            String[] parts = line.split(COMMA);
+            if (parts.length == NUMBER_OF_PARTS) {
+                String operationType = parts[0];
```

Comment:
If you see a comment like 'magic number' check all code base to resolve it. parts[0]; and parts[1]); are magic numbers as well